### ISSUE TITLE
Unhandled invalid WebApp name 

### ISSUE THREAD

**Issue by andreujuanc** [pulumi/pulumi-azure-nextgen#279]
Hello, 

I was creating a simple webapp and found this issue when a name already exists:

```
 pulumi:pulumi:Stack notification-service-development running 'dotnet build -nologo .' completed successfully
    azure-nextgen:resources/latest:ResourceGroup notification-service-dev  
    azure-nextgen:web/latest:AppServicePlan notification-service-dev  
 +  azure-nextgen:web/latest:WebApp notification-service-dev creating 
 +  azure-nextgen:web/latest:WebApp notification-service-dev creating error: autorest/azure: Service returned an error. Status=<nil> <nil>
 +  azure-nextgen:web/latest:WebApp notification-service-dev **creating failed** error: autorest/azure: Service returned an error. Status=<nil> <nil>

```

Code to create it is 1:1 the one from the examples. 
Using 0.* so i guess it's using v0.6.1-alpha.1612969235 

Adding random chars or changing the name to something unique fixes the issue.  Just the error result does not help mitigate 😿 

Thanks

**Comment by mikhailshilkov** [comment]
Yeah, I feel you https://twitter.com/MikhailShilkov/status/1354073195314376705

I guess we could be more explicit in name checks. Also, [auto-naming](https://github.com/pulumi/pulumi-azure-nextgen/issues/5) should help with getting examples more robust.

**Comment by andreujuanc** [comment]
So this is Azure returning nil? (╯°□°）╯︵ ┻━┻ 

Is there anyway the runtime could figure out what's the underlying issue might be?
Can we escalate this to Azure?

**Comment by mikhailshilkov** [comment]
> So this is Azure returning nil? 

Yes, they return this "Service returned an error. Status=<nil> <nil>".

> Is there anyway the runtime could figure out what's the underlying issue might be?

The actual error is visible in the activity log of a resource group. In theory, we could look it up there, but that's a long shot. Otherwise, it's mostly about preventing an invalid operation with more pre-validation.

> Can we escalate this to Azure?

You can try :) We don't have an efficient channel to service teams (this is not a problem of ARM, this is a problem of App Service).

**Comment by andreujuanc** [comment]
Pre-validation could help but there is so much that could be done. Getting the proper data would be best.
Since almost all deployments are on a resource group, would it be possible to show "statusMessage" on all failed runs? Maybe adding some verbosity argument to the cli.
